lib/p2p/Peer.ts

```diff
@@ -23,6 +23,23 @@ type HandshakeState = {
   pairs?: string[];
 };
 
+declare interface Peer {
+  on(event: 'packet', listener: (packet: Packet) => void): this;
+  on(event: 'error', listener: (err: Error) => void): this;
+  on(event: string, listener: Function): this;
+  once(event: 'open', listener: (handshakeState: HandshakeState) => void);
+  once(event: 'close', listener: () => void);
+  once(event: 'ban', listener: () => void);
+  once(event: string, listener: Function): this;
+  emit(event: 'connect');
+  emit(event: 'ban');
+  emit(event: 'open', handshakeState: HandshakeState);
+  emit(event: 'close');
+  emit(event: 'error', err: Error);
+  emit(event: 'packet', packet: Packet);
+  emit(event: string, ...any);
```

Comment:
what's this one for?

Reply:
I debated leaving that in - maybe it should be better if left out. This line (and line 29 and 33) allows for generic events without specific types. Basically this says you can `emit` anything where the first parameter is a string plus as many additional parameters of any type. Dropping this line would require anyone adding a new event type to come here and add new line defining the types it will use - and that might be a good thing after all. Let me know what you think.

Reply:
Yes I think it would be preferred, because if it's not enforced, we might forget adding a new event type.

Reply:
Agreed, removing them now.

Reply:
You can merge after settling this (either dropping these lines or not).